keras_cv/models/generative/stable_diffusion/stable_diffusion.py

```diff
@@ -89,10 +90,15 @@ def __init__(self, img_height=512, img_width=512, jit_compile=False):
         self.text_encoder = TextEncoder(MAX_PROMPT_LENGTH)
         self.diffusion_model = DiffusionModel(img_height, img_width, MAX_PROMPT_LENGTH)
         self.decoder = Decoder(img_height, img_width)
+
+        # TODO(lukewood): should we support variable size image inputs here?
+        self.image_encoder = VAEEncoder()
```

Comment:
This model will only be used for some use cases but creating it here will induce time and RAM overhead for all users. We should switch to a lazy model creation setup imo, something like `create_image_encoder()` called at the start of the method that requires it

Reply:
we'll likely just use the image encoder directly in the tutorial.  Perhaps a lazy getter would work?

Re: `ImageEncoder` SGTM.